work/v1/types.go

```diff
@@ -66,6 +70,65 @@ type DeleteOption struct {
 	SelectivelyOrphan *SelectivelyOrphan `json:"selectivelyOrphans,omitempty"`
 }
 
+// ManifestConfigOption represents the configurations of a manifest defined in workload field.
+type ManifestConfigOption struct {
+	// ResourceIdentifier represents the group, resource, name and namespace of a resoure.
+	// +kubebuilder:validation:Required
+	// +required
+	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
+
+	// FeedbackRules defines what resource status field should be returned.
+	// +kubebuilder:validation:Required
+	// +required
+	FeedbackRules []FeedbackRule `json:"feedbackRules"`
+}
+
+type FeedbackRule struct {
+	// Type defines the option of how status can be returned.
+	// It can be jsonPaths only or wellKnownStatus.
+	// If the type is JsonPaths, user should specify the jsonPaths field
+	// If the type is WellKnownStatus, certain common fields of status will be reported,
+	// including replicas, readyReplicas, and availableReplicas. And if these status fields
+	// do not exist, no values will be reported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type FeedBackType `json:"type"`
+
+	// JsonPaths defines the json path under status field to be synced.
+	// +optional
+	JsonPaths []JsonPath `json:"jsonPaths,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=WellKnownStatus;JsonPaths
+type FeedBackType string
+
+const (
+	// WellKnownStatusType represents that values of some common status fields will be returned, which
+	// include replica, readyReplica and availableReplica.
+	WellKnownStatusType FeedBackType = "WellKnownStatus"
```

Comment:
IIUC, the `WellKnownStatus` is actually a pre-built template of interested fields for reporting. if so, will the reporting fields differ depends on the resource type? e.g. if we apply `WellKnownStatus` to a job resource, then the reporting fields should be `.status.succeeded` etc. while if it's a deployment then `.spec.replicas` will be reported?

Reply:
I think this is similar as the comment here https://github.com/open-cluster-management-io/api/pull/108#discussion_r763066481 ^^^.  I think we should consider a WellKnownStatus per type or make it configurable in the long term. Maybe we start with a small list and if an item in the list cannot be found in status, we just skip it?

Reply:
> Maybe we start with a small list and if an item in the list cannot be found in status, we just skip it?

I'm ok with this. Perhaps with a rule that hardcoded items will only be for types in k8s.io/api and open-cluster-management/api, so we don't get vendor specific bits hardcoded?

Reply:
Updated the comment for this type